x-pack/plugins/apm/public/components/app/mobile/service_overview/filters/index.tsx

```diff
@@ -86,25 +93,29 @@ export function MobileFilters({
       responsive={false}
       direction={groupDirection}
     >
-      {data.mobileFilters.map((filter) => {
-        return (
-          <EuiFlexItem
-            grow={false}
-            key={filter.key}
-            style={isLarge ? {} : { width: '225px' }}
-          >
-            <EuiSelect
-              fullWidth={isSmall}
-              prepend={filter.label}
-              options={toSelectOptions(filter.options)}
-              value={filters[filter.key]}
-              onChange={(e) => {
-                onChangeFilter(filter.key, e.target.value);
-              }}
-            />
-          </EuiFlexItem>
-        );
-      })}
+      {isPending(status) ? (
+        <EuiLoadingSpinner size="m" />
```

Comment:
I couldn't see the loading status on the screen recording, did you add it after?

Reply:
No, it's there

but I'm not happy with the spinner but Eui have specific loading components

Reply:
I saw it! on the right corner
There's a prop `isLoading` available for the` EuiSelect`, maybe you can try to use that, also it might look nicer than having the filters just appearing from nothing, what do you think?

Reply:
https://user-images.githubusercontent.com/31922082/209102700-21f6e17a-9e17-46b1-bc4f-2cf23925e525.mov

Reply:
good point. 👍 

atm server is responsible for returning the filters, I'll need to change a bit but I think the UI will look better with your suggestion.

Reply:
@MiriamAparicio fixed it here: https://github.com/elastic/kibana/pull/147804/commits
Updated the recording with the latest changes